run_tests.sh

```diff
@@ -12,23 +12,25 @@ yum install -y golang qpid-proton-c-devel iproute
 go get -u golang.org/x/tools/cmd/cover
 go get -u github.com/mattn/goveralls
 go get -u golang.org/x/lint/golint
-go get -u github.com/golang/dep
-sh $GOPATH/src/github.com/golang/dep/install.sh
-
-# run test suite
+go get -u github.com/golang/dep/...
 dep ensure -v --vendor-only
-go test -v -timeout=10s ./tests/* || exit 1
 
-# run lints
-go vet ./cmd/main.go || exit 1
-golint . | xargs -r false || exit 1
+# run code validation tools
+echo " *** Running pre-commit code validation"
+./pre-commit
+
+# run unit tests
+echo " *** Running test suite"
+go test -v ./...
```

Comment:
This expands the places we can store tests?

Reply:
Yea I was testing it locally, and it looks like it searches the entire code base. If I run `go test -v ./...` and `go test -v tests/internal_pkg/*` the results are the same.

I think the former is better though, since it'll result in calling through the `main.go` as I was testing things locally and that seems to be what it's doing.

To be safe I could also just add both `go test -v ./...` and `go test -v tests/internal_pkg/*`.